### ISSUE TITLE
`Routify` app still has blocker to using `bun --bun vite`

### ISSUE THREAD

**Issue by hyuzea** [oven-sh/bun#3735]
### What version of Bun is running?

0.7

### What platform is your computer?

ubuntu 22.04

### What steps can reproduce the bug?

bun --bun vite

### What is the expected behavior?

After fix

![image](https://github.com/oven-sh/bun/assets/3823496/e0e82e80-3901-4471-aefa-02b29837f35a)

### What do you see instead?

here is my debug.

#### 1. call to `fs.lstatSync()` 
```
error: SystemError
 syscall: "TODO"
   errno: -254

      at /data/www/00-cup/bun-play/viteapp/node_modules/graceful-fs/polyfills.js:330:28
      at /data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:28:13
      at getStatsSync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:34:14)
      at checkPathsSync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:81:32)
      at copySync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:27:32)
```
```
SystemError: 
    at lstatSync (native)
    at <anonymous> (/data/www/00-cup/bun-play/viteapp/node_modules/graceful-fs/polyfills.js:331:17)
    at <anonymous> (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:27:48)
    at getStatsSync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:40:4)
    at checkPathsSync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/util/stat.js:82:16)
    at copySync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:27:42)
```
`fs-extra` sets `bigint: true` option which seems not available in bun, set to `undefined` will fix the issue
```
/data/www/00-cup/bun-play/viteapp/node_modules/graceful-fs/polyfills.js#statFixSync {
  orig: [Function: lstatSync],
  target: "/data/www/00-cup/bun-play/viteapp/node_modules/@roxi/routify/lib/buildtime/plugins/filemapper/lib/utils/[...404].svelte",
  options: {
    bigint: true
  }
}
```

#### 2. call to `fs.chmodSync()` -- Resolved by: #3747 
```
TypeError: Invalid mode: must be an octal number
    at chmodSync (native)
    at <anonymous> (/data/www/00-cup/bun-play/viteapp/node_modules/graceful-fs/polyfills.js:269:8)
    at setDestMode (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:102:35)
    at copyFile (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:82:39)
    at onFile (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:59:9)
    at getStats (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:51:11)
    at handleFilterAndCopy (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:35:43)
    at copySync (/data/www/00-cup/bun-play/viteapp/node_modules/fs-extra/lib/copy/copy-sync.js:28:54)
```

`fs.stat()` returns `stat.mode` which has integer type. set `mode = mode.toString(8).slice(2)` before `fs.chmodSync()` calls will fixs the issue, in example its converts `33188` to `"0644"`  
```
/data/www/00-cup/bun-play/viteapp/node_modules/graceful-fs/polyfills.js#chmodFixSync {
  orig: [Function: chmodSync],
  target: ".routify/components/[...404].svelte",
  mode: 33188
}
```

**Comment by hyuzea** [comment]
`chmodSync()` issue may related with #3740

**Comment by paperclover** [comment]
lstatSync will be resolved by #4208

**Comment by hyuzea** [comment]
its worked
